src/uri/uri.cc

```diff
@@ -249,6 +249,34 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::string{path}};
+  const std::string path_to_set = parsed_path.path().value();
+  const bool has_leading_slash = path_to_set.front() == '/';
+
+  this->path_ = has_leading_slash ? path_to_set.substr(1) : path_to_set;
+  return *this;
+}
+
+auto URI::path(std::string &&path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::move(path)};
```

Comment:
You probably don't want this to be `const`, as otherwise you still do another copy